vendor/bat-native-ledger/src/bat/ledger/internal/media/twitter.cc

```diff
@@ -63,6 +67,120 @@ std::string MediaTwitter::GetMediaKey(const std::string& screen_name) {
   return (std::string)TWITTER_MEDIA_TYPE + "_" + screen_name;
 }
 
+// static
+std::string MediaTwitter::GetUserNameFromUrl(const std::string& path) {
+  if (path.empty()) {
+    return std::string();
+  }
+
+  std::vector<std::string> parts = base::SplitString(
+      path, "/", base::TRIM_WHITESPACE, base::SPLIT_WANT_NONEMPTY);
+
+  if (parts.size() > 0) {
+    return parts.at(0);
+  }
+
+  return std::string();
+}
+
+// static
+bool MediaTwitter::IsExcludedPath(const std::string& path) {
+  if (path.empty()) {
+    return true;
+  }
+
+  const std::vector<std::string> paths({
+      "/",
+      "/settings",
+      "/explore",
+      "/notifications",
+      "/messages",
+      "/logout",
+      "/search",
+      "/about",
+      "/tos",
+      "/privacy",
+      "/home"
+    });
+
+  for (std::string str_path : paths) {
+    if (str_path == path || str_path + "/" == path) {
+      return true;
+    }
+  }
+
+  const std::vector<std::string> patterns({
+    "/i/",
+    "/account/",
+    "/compose/",
+    "/who_to_follow/",
+    "/hashtag/",
+    "/settings/"
+  });
+
+  for (std::string str_path : patterns) {
+    if (base::StartsWith(path,
+                         str_path,
+                         base::CompareCase::INSENSITIVE_ASCII)) {
+      return true;
+    }
+  }
+
+
+  return false;
+}
+
+// static
+std::string MediaTwitter::GetUserId(const std::string& response) {
+  if (response.empty()) {
+    return std::string();
+  }
+
+  std::string id = braveledger_media::ExtractData(
+      response,
+      "<div class=\"ProfileNav\" role=\"navigation\" data-user-id=\"", "\">");
+
+  if (id.empty()) {
+    id = braveledger_media::ExtractData(
+      response,
+      "https://pbs.twimg.com/profile_banners/", "/");
+  }
+
+  return id;
+}
+
+// static
+std::string MediaTwitter::GetPublisherName(const std::string& response,
+                                           const std::string& user_name) {
+  if (response.empty()) {
+    return std::string();
+  }
+
+  std::string title = braveledger_media::ExtractData(
```

Comment:
For anything we extract from untrusted content it might be nice to label variable names with `unsafe` or similar as a flag that this data can't be used without sanitization anywhere. Example in SQL statements without bind, in HTML snippets, etc.
Basically untrusted content can have things like `<script>` in it and `"; DROP TABLE;` etc.

Then if you combine any other data with this data you'd also have to have similar naming. Example as you build a URL with it via `sprintf` then that itself becomes unsafe data.

Reply:
`ExtractData` is I think always used to get things from HTML code that is untrusted content. So I think we can add `sanitization` into `ExtractData` and also in `SaveVisit` just to be double sure

Reply:
Is it maybe better to only extract alphanumeric chars via a regex or something like that?  Should we sanitize early?  Punycode? 
cc @diracdeltas for best practices for untrusted content.

Reply:
seems like maybe `braveledger_media::ExtractData` might be a good place to add the sanitization.

Reply:
I was just looking at this last week and thinking the same thing. seems best to ensure output is only alphanumeric if that is acceptable.